nuxeo-routing-core/src/main/java/org/nuxeo/ecm/platform/routing/core/io/TaskWriter.java

```diff
@@ -147,42 +148,13 @@ public void writeEntityBody(Task item, JsonGenerator jg) throws IOException {
             }
             jg.writeEndArray();
 
-            jg.writeArrayFieldStart("actors");
             final boolean isFetchActors = ctx.getFetched(ENTITY_TYPE).contains(FETCH_ACTORS);
-            for (String actorId : item.getActors()) {
-                if (isFetchActors) {
-                    if (actorId.startsWith(USER_PREFIX + SEPARATOR)) {
-                        actorId = actorId.substring(USER_PREFIX.length() + SEPARATOR.length());
-                        NuxeoPrincipal user = userManager.getPrincipal(actorId);
-                        if (user != null) {
-                            writeEntity(user, jg);
-                            continue;
-                        }
-                    } else if (actorId.startsWith(GROUP_PREFIX + SEPARATOR)) {
-                        actorId = actorId.substring(GROUP_PREFIX.length() + SEPARATOR.length());
-                        NuxeoGroup group = userManager.getGroup(actorId);
-                        if (group != null) {
-                            writeEntity(group, jg);
-                            continue;
-                        }
-                    } else {
-                        NuxeoPrincipal user = userManager.getPrincipal(actorId);
-                        if (user != null) {
-                            writeEntity(user, jg);
-                            continue;
-                        } else {
-                            NuxeoGroup group = userManager.getGroup(actorId);
-                            if (group != null) {
-                                writeEntity(group, jg);
-                                continue;
-                            }
-                        }
-                    }
-                }
-                jg.writeStartObject();
-                jg.writeStringField("id", actorId);
-                jg.writeEndObject();
-            }
+            jg.writeArrayFieldStart("actors");
+            writeActors(item.getActors(), isFetchActors, jg);
+            jg.writeEndArray();
+
+            jg.writeArrayFieldStart("delegatedActors");
+            writeActors(item.getDelegatedActors(), isFetchActors, jg);
```

Comment:
This is refactoring which is nice but can you please make a separate commit to isolate it from the changes purely related to the reassignment issue.